bin/grommet.js

```diff
@@ -17,76 +18,120 @@ String.prototype.capitalize = function() {
-      console.error(
-        '[grommet] Currently you have Node ' + process.version +
-        ' and NPM ' + npmVersion
-      );
-      process.exit(1);
-    }
+function dependenciesSupported(nodeVersion, npmVersion) {
+  if (!nodeVersionSupported(nodeVersion) || !npmVersionSupported(npmVersion)) {
+    console.error(
+      '[grommet] Grommet requires Node v'+supportedNodeVersion+
+      '+ and NPM '+supportedNpmVersion+'+.');
+    console.error(
+      '[grommet] Currently you have Node ' + process.version +
+      ' and NPM ' + npmVersion
+    );
+    return false;
+  }
+
+  return true;
+}
+
+function copyAssetsAndInstall(from, to, command) {
+  var installDependencies = function() {
+    console.log("[grommet] Installing Grommet dependencies...");
+    return install();
+  }
 
-    mkdirp('./' + app, function(err) {
-      if (err) {
-        console.log('Error trying to create project: ' + err);
-      } else {
-        process.chdir('./' + app);
-
-        var templateFolder = path.join(grommetPath, 'templates/init/**');
-        var mobileIcon = path.join(grommetPath, 'img/mobile-app-icon.png');
-        var shortcutIcon = path.join(grommetPath, 'img/shortcut-icon.png');
-
-        gulp.src(mobileIcon).pipe(gulp.dest('./src/img'));
-        gulp.src(shortcutIcon).pipe(gulp.dest('./src/img'));
-        gulp.src(templateFolder, { dot: true })
-        .pipe(template({
-          appName: app,
-          appTitle: title,
-          grommetVersion: packageJSON.version
-        }))
-        .pipe(gulp.dest('./'))
-        .pipe(install()).on('finish', function() {
-          done();
-        });
+  var templateVars = {
+    appName: command.options.app,
+    appTitle: command.options.title,
+    appDescription: command.options.description,
+    grommetVersion: command.env.grommetVersion
+  };
+
+  gulp.src(from, { dot: true })
+    .pipe(template(templateVars))
+    .pipe(gulp.dest(to))
+    .pipe(installDependencies());
+}
+
+var tasks = {
+  'init': function(command) {
+    return function() {
+      if (!dependenciesSupported(command.env.nodeVersion, command.env.npmVersion)) {
+        process.exit(1);
       }
-    });
+
+      var newAppPath = path.join(process.cwd(), command.options.app);
+
+      mkdirp(newAppPath, function(err) {
+        if (err) {
+          console.log('Error trying to create project: ' + err);
+        } else {
+          var templateFolder = path.join(command.env.templatePath, 'init/**');
+          var mobileIcon = path.join(command.env.grommetPath, 'img/mobile-app-icon.png');
+          var shortcutIcon = path.join(command.env.grommetPath, 'img/shortcut-icon.png');
+
+          gulp.src(mobileIcon).pipe(gulp.dest('./src/img'));
+          gulp.src(shortcutIcon).pipe(gulp.dest('./src/img'));
+          copyAssetsAndInstall(templateFolder, newAppPath, command);
+        }
+      })
+    }
   }
 };
 
+var supportedNodeVersion = '0.10';
+var supportedNpmVersion = '1.4';
```

Comment:
We recently updated our support stack to 3+ for NPM. would you be kind enough to update here ?